app/src/main/java/org/simple/clinic/mobius/MobiusDelegate.kt

```diff
@@ -85,7 +85,13 @@ class MobiusDelegate<M : Parcelable, E, F>(
     if (::eventsDisposable.isInitialized && eventsDisposable.isDisposed.not()) {
       eventsDisposable.dispose()
     }
-    controller.stop()
+    with(controller) {
+      if (isRunning.not()) {
+        start()
+      }
+      stop()
+      disconnect()
+    }
```

Comment:
Since `MobiusDelegate` already has `start()` and `stop()` functions, it is not *obvious* that the `start()` and `stop()` functions here are actually methods on the `Controller`. In addition, this could potentially cause problems since removing the `with()` clause won't cause compilation errors and will end up in a recursive function call.

Can we extract these into separate methods? Suggestions: `startControllerIfNotAlreadyRunning` and `stopAndDisconnectController`.

Reply:
Done!